File: aafixer.pl
#!/usr/bin/env perl
use strict;
use warnings;
use File::Find;
use Getopt::Long;
use Image::Size;
use feature qw{ say };

# Parse arguments
GetOptions(
    'input|i=s' => \my $input,
    'output|o=s' => \my $output
);
my %options = ( input => $input, output => $output );

# Ensure directory paths end in slashes
$options{input} .= '/' unless substr($options{input}, -1) =~ m/\//;
$options{output} .= '/' unless substr($options{output}, -1) =~ m/\//;

# Variable declaration
my @not_square;
my @not_big_enough;

# Sub to feed to File::Find sub
sub album_art_fixer
{
    my $file = $File::Find::name;

    if ($file =~ /AlbumArt\.jpg$/)
    {
        my %aadim;
        ($aadim{width}, $aadim{height}) = imgsize($file);

        if ($aadim{width} == $aadim{height})
        {
            push @not_square, $file;
        }

        if (($aadim{width} > 1000) || ($aadim{height} > 1000))
        {
            push @not_big_enough, $file;
        }
    }
}

# Main part of script
find(\&album_art_fixer, $options{input});
my %file_paths = ( notbig => $options{output} . 'notbig.txt', notsquare => $options{output} . 'notsquare.txt' );

open(my $bfh, '>', $file_paths{notbig});
foreach (@not_big_enough)
{
    say $bfh "$_";
}
close $bfh;

open(my $sfh, '>', $file_paths{notsquare});
foreach (@not_square)
{
    say $sfh "$_";
}
close $sfh;

File: aafixer.jl
#!/usr/bin/env julia
# Parse arguments
inputpath = ARGS[2]
outputpath = ARGS[4]

# Ensure directory paths end in slashes
if inputpath[end] != '/'
    inputpath = inputpath * "/"
end

if outputpath[end] != '/'
    outputpath = outputpath * "/"
end

# Variable declaration
notsquare = String[]
notbigenough = String[]


# Main part of script
for (root, dirs, files) in walkdir(inputpath)
    aarts = filter(x -> contains(x, "AlbumArt.jpg"), files)
    for f in aarts
        aadim = Dict{String,String}()
        ffpath = joinpath(root, f)
        result = readstring(`magick identify $ffpath`)
        dims = match(r"\d+x\d+", result)
        (aadim["width"], aadim["height"]) = split(dims.match, "x")

        # Check if image is square
        if aadim["width"] != aadim["height"]
            push!(notsquare, ffpath)
        end

        # Check if image is big enough
        if parse(Int64, aadim["width"]) < 1000 || parse(Int64, aadim["height"]) < 1000
            push!(notbigenough, ffpath)
        end
    end
end

# Write results of search to files
writedlm(outputpath * "notbig.txt", notbigenough, "\n")
writedlm(outputpath * "notsquare.txt", notsquare, "\n")
